File: feature_storage/src/main/java/com/wang/mytest/feature/storage/database/DatabaseActivity.java
package com.wang.mytest.feature.storage.database;

import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.util.Log;

import com.wang.mytest.apt.annotation.Route;
import com.wang.mytest.feature.storage.R;

import androidx.annotation.Nullable;
import androidx.fragment.app.FragmentActivity;

@Route(path = "/activity/storage/database", title = "Database")
public class DatabaseActivity extends FragmentActivity {

    private static final String TAG = "DatabaseActivity";

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_database);

//        openDatabase();
    }

    private void openDatabase() {
        SQLiteDatabase db = openOrCreateDatabase("test.db", MODE_PRIVATE, null);
        db.execSQL("CREATE TABLE IF NOT EXISTS products("
                + "prod_id TEXT NOT NULL PRIMARY KEY, "
                + "vend_id TEXT NOT NULL, "
                + "prod_name TEXT NOT NULL, "
                + "prod_price DECIMAL NOT NULL, "
                + "prod_desc TEXT NOT NULL"
                + ");"
        );

        db.execSQL("INSERT INTO products(prod_id, vend_id, prod_name, prod_price, prod_desc) VALUES('ID001', 'DLL001', 'XiaoShuiKu', 22.2, 'PuShui')");

        db.execSQL("INSERT INTO products(prod_id, vend_id, prod_name, prod_price, prod_desc) VALUES(?,?,?,?,?)",
                new Object[] {"ID002", "DLL002", "XiaoShuiKu2", 55.2, "PuShui2"});

        db.execSQL("INSERT INTO products(prod_id, vend_id, prod_name, prod_price, prod_desc) VALUES(?,?,?,?,?)",
                new Object[] {"ID003", "DLL003", "XiaoShuiKu3", 70.2, "PuShui3"});

        Cursor cursor = db.rawQuery("SELECT prod_id, prod_name, prod_price, prod_desc FROM products WHERE prod_price BETWEEN ? AND ?",
                new String[]{"22.2", "55.4"});
        Log.d(TAG, "onCreate: cursor = " + cursor);
        if (cursor != null) {
            while (cursor.moveToNext()) {
                String prod_id = cursor.getString(cursor.getColumnIndex("prod_id"));
                String prod_name = cursor.getString(cursor.getColumnIndex("prod_name"));
                double prod_price = cursor.getDouble(cursor.getColumnIndex("prod_price"));
                String prod_desc = cursor.getString(cursor.getColumnIndex("prod_desc"));
                Log.d(TAG, "onCreate: prod_id = " + prod_id);
                Log.d(TAG, "onCreate: prod_name = " + prod_name);
                Log.d(TAG, "onCreate: prod_price = " + prod_price);
                Log.d(TAG, "onCreate: prod_desc = " + prod_desc);
            }
        }
    }
}

File: feature_storage/src/main/java/com/wang/mytest/feature/storage/database/DatabaseActivity.kt
package com.wang.mytest.feature.storage.database

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import androidx.viewpager2.widget.ViewPager2
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.wang.mytest.feature.storage.R

/**
 * BottomNavigationView + ViewPager2 + RecyclerView
 */
@Suppress("UNREACHABLE_CODE")
class DatabaseActivity : FragmentActivity() {

    companion object {
        private const val TAG = "DatabaseActivity"
        private const val ITEM_POSITION_QUERY = 0
        private const val ITEM_POSITION_INSERT = 1
        private const val ITEM_COUNT = 2
    }

    private lateinit var viewPager: ViewPager2
    private lateinit var bottomNavigation: BottomNavigationView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_database)

        viewPager = findViewById<ViewPager2>(R.id.view_pager).apply {
            adapter = MyAdapter(this@DatabaseActivity)
            registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
                override fun onPageSelected(position: Int) {
                    Log.d(TAG, "onPageSelected: position = $position");
                }
            })
        }

        bottomNavigation = findViewById<BottomNavigationView>(R.id.bottom_navigation).apply {
            setOnNavigationItemSelectedListener {
                when (it.itemId) {
                    R.id.navigation_previous -> {
                        viewPager.currentItem = ITEM_POSITION_QUERY
                    }
                    R.id.navigation_next -> {
                        viewPager.currentItem = ITEM_POSITION_INSERT
                    }
                }
                true
            }
        }
    }

    class MyAdapter(fragmentActivity: FragmentActivity) : FragmentStateAdapter(fragmentActivity) {
        override fun getItemCount(): Int = ITEM_COUNT

        override fun createFragment(position: Int): Fragment {
            return when (position) {
                ITEM_POSITION_QUERY -> DatabaseQueryFragment()
                ITEM_POSITION_INSERT -> DatabaseInsertFragment()
                else -> Fragment()
            }
        }
    }
}
